test/e2e/app-dir/use-cache/use-cache.test.ts

```diff
@@ -241,4 +241,29 @@ describe('use-cache', () => {
       expect(await browser.elementByCss('p').text()).toBe('result')
     })
   })
+
+  it('should be able to revalidate a page using', async () => {
+    const browser = await next.browser(`/form`)
+    const time1 = await browser.waitForElementByCss('#t').text()
+
+    await browser.loadPage(new URL(`/form`, next.url).toString())
+
+    const time2 = await browser.waitForElementByCss('#t').text()
+
+    expect(time1).toBe(time2)
+
+    await browser.elementByCss('#refresh').click()
+
+    await waitFor(500)
```

Comment:
This may have the potential to become flaky. We could do this instead:
```ts
await retry(async () => {
  const time3 = await browser.waitForElementByCss('#t').text()
  expect(time3).not.toBe(time2)
})
```

Or are the 500ms related to the revalidation time?

Reply:
no it really should be made into something more like wait for a dom node to indicate that the action has completed but I couldn't figure out how to do it without a way to using useActionState.